### PR TITLE
Extend persistent storage tests

### LINKED ISSUE #743: Run all existing tests against real storages too in the CI
We should try just running all the existing tests against real storages in the CI and checking the cost is reasonable, since it hasn't been very expensive so far.

We can do it by swapping out the s3 fixture used from the simulator to real storage, in a dedicated (once a day?) job.

### PR DESCRIPTION
**Pull request by G-D-Petrov**
#### Reference Issues/PRs
Fixes #743 

#### What does this implement/fix? How does it work (high level)? Highlight notable design decisions.
Extends the persistent stores to most stress and integration tests

### CODE REVIEW

File: python/tests/stress/arcticdb/version_store/test_stress_append.py
Comment (poodlewars) [inline python/tests/stress/arcticdb/version_store/test_stress_append.py:100]: `big_map` is an LMDB specific concept. But think the idea is fine. Just calling the fixture `basic_store_large_data` or something would be fine I think.

File: python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py
Comment (poodlewars) [inline python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py:18]: Yeah, we shouldn't need to skip Azure for this. Good spot!

File: python/tests/integration/arcticdb/test_arctic.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/test_arctic.py:72]: Hm, seems like a shame. Could get the `arctic_client` to clean itself up after each test?

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:401]: Def want to test this one with both symbol list on and symbol list off.

File: python/tests/stress/arcticdb/version_store/test_stress_version_map_compact.py
Comment (poodlewars) [inline python/tests/stress/arcticdb/version_store/test_stress_version_map_compact.py:1]: Same comment as on your code coverage PR, would be good to explain why we are fiddling around with this test.

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (G-D-Petrov) [reply python/tests/integration/arcticdb/version_store/test_basic_version_store.py:401]: The comment was more-so if we to test no_symbol_list fixture for s3, real-s3, etc....

File: python/tests/util/date.py
Comment (poodlewars) [inline python/tests/util/date.py:1]: Do not merge this change

File: python/tests/unit/arcticdb/version_store/test_parallel.py
Comment (poodlewars) [inline python/tests/unit/arcticdb/version_store/test_parallel.py:1]: Do not merge this change

**Comment by poodlewars** [comment]
Looking promising! For future cross-cutting changes like this, please could we make it a bit more reviewable? I like to do this with commits that do each rename one by one. Then the reviewer can just glance through the commit and trust that the IDE basically did the job correctly.

File: python/tests/integration/arcticdb/test_arctic.py
Comment (G-D-Petrov) [reply python/tests/integration/arcticdb/test_arctic.py:72]: Yeah I will look into this

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:929]: Scary! Why do we need this?

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/version_store/test_basic_version_store.py:1]: Does this run against more test fixtures even when running locally now?

Keen not to slow down local testing any more than we have to

File: python/tests/integration/arcticdb/version_store/test_symbol_list.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/version_store/test_symbol_list.py:101]: Scary! What's going on here?

File: python/tests/integration/arcticdb/version_store/test_symbol_list.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/version_store/test_symbol_list.py:199]: Could you raise a github ticket for this? Good spot

**Review by poodlewars — APPROVED** [review]

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (G-D-Petrov) [reply python/tests/integration/arcticdb/version_store/test_basic_version_store.py:1]: Potentially, there might be a few more runs, if azure_support is enabled.
Shouldn't be noticeable

File: python/tests/integration/arcticdb/version_store/test_basic_version_store.py
Comment (G-D-Petrov) [reply python/tests/integration/arcticdb/version_store/test_basic_version_store.py:929]: Will remove

File: python/tests/integration/arcticdb/test_arctic.py
Comment (G-D-Petrov) [reply python/tests/integration/arcticdb/test_arctic.py:236]: test_separation_between_libraries - tests exactly this

File: python/arcticdb/version_store/helper.py
Comment (G-D-Petrov) [reply python/arcticdb/version_store/helper.py:300]: we expect them to show, test_separation_between_libraries test this and it is passing

File: python/tests/stress/arcticdb/version_store/test_stress_delete.py
Comment (G-D-Petrov) [reply python/tests/stress/arcticdb/version_store/test_stress_delete.py:27]: Will remove

**Review by poodlewars — APPROVED** [review]
Looks good!

Please squash the commits before merging this one.

File: build_tooling/parallel_test.sh
Comment (poodlewars) [inline build_tooling/parallel_test.sh:10]:
```suggestion
# If the persistent storage tests are not enabled, we want to parallelise the pytests
```

File: build_tooling/parallel_test.sh
Comment (poodlewars) [inline build_tooling/parallel_test.sh:14]:
```suggestion
# If they are enabled, we want to run the pytests sequentially to avoid races
```

File: python/arcticdb/util/storage_test.py
Comment (poodlewars) [inline python/arcticdb/util/storage_test.py:28]: No need for the ternary operator here.

```suggestion
    clear = str(os.getenv("ARCTICDB_REAL_S3_CLEAR")).lower() in ("true", "1")
```

File: python/arcticdb/util/storage_test.py
Comment (poodlewars) [inline python/arcticdb/util/storage_test.py:70]: Might as well use `assert_frame_equal`

File: python/arcticdb/util/storage_test.py
Comment (poodlewars) [inline python/arcticdb/util/storage_test.py:144]: `AST` is a Man Group thing, let's use dummy ISIN codes https://en.wikipedia.org/wiki/International_Securities_Identification_Number

File: python/arcticdb/util/storage_test.py
Comment (poodlewars) [inline python/arcticdb/util/storage_test.py:60]: More specific name for this and `seed_library` might be good, as they're both only useful for the little 3 col dataframes

File: python/arcticdb/util/storage_test.py
Comment (poodlewars) [inline python/arcticdb/util/storage_test.py:154]: Comment is wrong? Sum will be `n(n+1)/2`?

File: python/tests/integration/arcticdb/test_persistent_storage.py
Comment (poodlewars) [inline python/tests/integration/arcticdb/test_persistent_storage.py:87]: It's deterministic, so can we just use `assert_frame_equal`?

File: build_tooling/get_newest_latest_version.py
Comment (G-D-Petrov) [reply build_tooling/get_newest_latest_version.py:1]: the idea of the script is to get the latest major version automatically get "oldest" version for the "newest" major version that we have. for example:
- 3.0.0 if we are on 3.x.x
- 4.0.0 if we are on 4.x.x
- etc

File: python/arcticdb/util/storage_test.py
Comment (G-D-Petrov) [reply python/arcticdb/util/storage_test.py:144]: I wanted to have just some string that changes deterministically, so I would prefer something like TK_{..} or something.
Is that okay?

File: .github/workflows/persistent_storage.yml
Comment (poodlewars) [inline .github/workflows/persistent_storage.yml:58]: Might be good to explain what "supported" means here, ie that we're talking about forwards compatibility and we expect libs written by master only to be readable by arcticdb>=3.0.0

File: python/tests/util/storage_test.py
Comment (poodlewars) [inline python/tests/util/storage_test.py:28]: Missing this suggestion https://github.com/man-group/ArcticDB/pull/789#discussion_r1328955491

File: python/tests/util/storage_test.py
Comment (poodlewars) [inline python/tests/util/storage_test.py:154]: Comment is wrong? The values aren't random, and their sum isn't n

File: python/tests/util/storage_test.py
Comment (poodlewars) [inline python/tests/util/storage_test.py:229]: Can you make a github ticket to do Azure?

**Review by poodlewars — CHANGES_REQUESTED** [review]

**Comment by poodlewars** [comment]
- Rebase and squash commits

File: python/tests/util/storage_test.py
Comment (G-D-Petrov) [reply python/tests/util/storage_test.py:229]: Done - issue #876

**Review by poodlewars — APPROVED** [review]

**Pull request merged**
